Documentation/Building.md

```diff
@@ -1,11 +1,64 @@
 # Building and installing Libevent
 ### Jump to:
+- [Prerequisites](#Prerequisites)
 - [Autoconf](#autoconf)
   - [Flags](#autoconf-flags)
 - [Building on Windows](#building-on-windows)
 - [Building on Unix (With CMake)](#building-on-unix-cmake)
 - [CMake Variables](#cmake-variables)
 
+## Prerequisites
+
+### Linux deb-like (ubuntu/debian/...)
+Just install tools using your preferred package manager if you build using autotools:
+```sh
+sudo apt-get install automake autoconf libtool pkg-config
+```
+or build using cmake:
+```sh
+sudo apt-get install cmake
+```
+Doxygen is a tool for generating documentation. Git is used to fetch the package version.  
+Install them if needed:
+```
+sudo apt-get install doxygen git
+```
+It is highly recommended to support OpenSSL encryption:
+```sh
+sudo apt-get install libssl-dev
+```
+To support multithreaded environments, libpthread is a must, and it already exists in the system. 
+
+To run the tests, you should install zlib:
+```sh
+sudo apt-get install zlib1g-dev
+```
+Finaly, python interpreter should be installed if you want to run regress tests.
+
+### MacOS
+It is recommended to use brew to install software on MacOS.
```

Comment:
I would say that "brew can be used" or "you can use brew", since libevent do not have any requirement from brew, except for common sense, i.e. `make install` is very-very-bad pattern